src/Common/StringParsing.fs

```diff
@@ -297,15 +307,16 @@ module StringPosition =
 
     /// Matches when a string consists of some number of
     /// complete repetitions of a specified sub-string.
+    [<return: Struct>]
     let (|EqualsRepeated|_|) (repeated, _n: MarkdownRange) =
         function
-        | StartsWithRepeated repeated (_n, (v, _)) when (String.IsNullOrWhiteSpace v) -> Some()
-        | _ -> None
+        | StartsWithRepeated repeated (_n, (v, _)) when (String.IsNullOrWhiteSpace v) -> ValueSome()
+        | _ -> ValueNone
 
 module List =
     /// Matches a list if it starts with a sub-list that is delimited
     /// using the specified delimiters. Returns a wrapped list and the rest.
-    let inline (|DelimitedWith|_|) startl endl input =
+    let inline internal (|DelimitedWith|_|) startl endl input =
```

Comment:
So `internal` because this is used in other files in the project?
Otherwise `private` would also work?

Reply:
I just want to limit the scope to this solution: As you know inline makes compiler slower. So if you have hundreds of public inline functions in your solution and someone refers your project via Nuget package, they are going to pay the price with their compilation too, because those inline-functions come available for them too, even when they probably are not aware of them and probably will never need them.

Reply:
This is my reasoning only, and I might be wrong:
If do code like this: `myInt |> ` and now you ask your F# IDE to auto-complete, then when it knows the type of "myInt" then it can give proper function options: Those who take int in this case. 

But in case of inline functions, they use statically resolved type parameters.

So if you have a referenced 10s other nuget packages and all have 10s of inline functions, then... all of those might be potentials for your auto-complete. You probably don't want call those, so it's just noise that could be avoided if they'd be internal functions.